### ISSUE TITLE
ConcurrentClient uploads twice

### ISSUE THREAD

**Issue by andreapinti** [qusly/qusly-core#41]
When using upload method with **ConcurrentClient**, it is uploaded twice and the result file is 2x bigger than the local one.

No matters if you use 1 or 3 parallel clients.

Sample test

```javascript
const qusly = require('qusly-core');
async function test(){
    var client = new qusly.ConcurrentClient(1);

    await client.connect({
        host: "localhost",
        user: "me",
        // ...
        protocol: "ftp"
    });

    const localPath = "myLocalFile.txt";
    const remotePath = "myRemoteFile.txt";
    client.on("progress", (e,info) => {
        console.log("Percent:",e.percent);
    })
    await client.upload(
        localPath,
        remotePath
    )
    await client.disconnect();

}
test().then(r => console.log(r))
```

qusly-core version: **2.5.1**

**Comment by xnerhu** [comment]
I could reproduce this bug using `binary` mode, which is used to transfer files like videos or music.
It seems that both `ftp` and `sftp` are affected. This is another case where #40 would be very useful at detecting such bugs.

**Comment by xnerhu** [comment]
I've fixed it. Everything was fine with the lowest part, which is deciding what dependency needs to be used for a given protocol to transfer a file. See [`TransferManager`](https://github.com/qusly/qusly-core/blob/master/src/models/transfer-manager.ts#L77). The problem was with encoding option set to `utf8` in `createReadStream` and `createWriteStream` . See [`ConcurrentClient`](https://github.com/qusly/qusly-core/blob/master/src/models/concurrent-client.ts#L193).

**Comment by xnerhu** [comment]
Released in `v2.5.2`.
